### PR TITLE
Cookie kit bugs

### PR DESCRIPTION
**Pull request by maklyakov**

### CODE REVIEW

**Comment by maklyakov** [comment]
[Cookie Kit Bugs](https://trello.com/c/SlWphXYM/1922-cookie-kit-bugs)

**Comment by maklyakov** [comment]
[Cookie Kit - Flag sort](https://trello.com/c/hUbdcKXm/1917-cookie-kit-flag-sort)

**Comment by maklyakov** [comment]
[Cookie Kit detectCountry flag implemented incorrectly](https://trello.com/c/YHOvuU38/1918-cookie-kit-detectcountry-flag-implemented-incorrectly)

File: packages/xcoobee-cookie-kit-react/src/CookieKitContainer.js [hunk 1]
```diff
@@ -206,21 +206,25 @@ export default class CookieKitContainer extends React.PureComponent {
...
+    const defaultCountryCode = "EU";
+
     let promise = null;
 
     if (countryCode) {
       return Promise.resolve(countryCode);
     }
-    if (!!campaignReference && detectCountry) {
-      promise = Promise.resolve(fetchCountryCodeForSubscribers(campaignReference));
+    if (detectCountry) {
```
Comment (dhurlburtusa) [inline packages/xcoobee-cookie-kit-react/src/CookieKitContainer.js]: To make the code easier to follow, let's move the `let promise = null;` into this `if` statement. Then let's move the `return promise\n      .catch(...` after the following `if/else` statement. Then we can get rid of the `else` clause of this `if` statement and simply `return Promise.resolve(defaultCountryCode);`. This way, the `promise.catch(...).then(...);` is grouped together with the corresponding code branch.

So, something like:

```suggestion
    if (detectCountry) {
      let promise;
      if (campaignReference) {
        promise = Promise.resolve(fetchCountryCodeForSubscribers(campaignReference));
      } else {
        promise = Promise.resolve(fetchCountryCode());
      }
      return promise
        .catch((error) => {
          console.error(error);
          this.setState({ countryCode: defaultCountryCode });

          return defaultCountryCode;
        })
        .then((cCode) => {
          saveCountryCode(cCode);
          this.setState({ countryCode: cCode });
          return cCode;
        });
    }
    return Promise.resolve(defaultCountryCode);
```

**Pull request merged**